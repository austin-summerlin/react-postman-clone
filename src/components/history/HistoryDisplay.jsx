import React from 'react';
import PropTypes from 'prop-types';


const HistoryDisplay = ({ method, url, id }) => {
  return (
    <article id={id}>
      <p id={id}>{method}</p>
      <span id={id}>{url}</span>
    </article>
  );
};


HistoryDisplay.propTypes = {
  id: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
};

export default HistoryDisplay;

