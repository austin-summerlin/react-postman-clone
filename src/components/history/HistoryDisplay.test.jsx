/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import HistoryDisplay from '../history/HistoryDisplay';

describe('History Display Component', () => {
  const history = [{ 'url': '', 'method': 'PUT', 'body': '', 'key': '+PUT' }, { 'url': '', 'method': 'GET', 'body': 'stuff', 'key': '+GET' }];
  const onClick = jest.fn();
  afterEach(() => cleanup());
  it.only('should render History Display', () => {
    const { asFragment } = render(<HistoryDisplay history={history} onClick={onClick} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
