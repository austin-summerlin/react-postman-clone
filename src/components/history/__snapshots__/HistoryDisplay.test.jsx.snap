// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History Display Component should render History Display 1`] = `
<DocumentFragment>
  <article>
    <p />
    <span />
  </article>
</DocumentFragment>
`;
